addons/web/static/src/webclient/commands/command_palette.js

```diff
@@ -18,7 +18,11 @@ const commandCategoryRegistry = registry.category("command_categories");
 const commandProviderRegistry = registry.category("command_provider");
 
 /**
- * @typedef {import("./command_service").Command} Command
+ * @typedef {{
+ *  action: ()=>void,
+ *  category?: string,
+ *  props: Object,
```

Comment:
```suggestion
 *  Component?: owl.Component,
 *  props?: Object,
```

Reply:
As `Component` is optional, `props` should be optional too.

Thus the need, IMHO, to pass `name` outside of the `props` scope...

And then pass `commandItem.name` in the command palette template either as a prop:
```xml
<t t-component="commandItem.Component" name="commandItem.name" t-props="commandItem.props"/>
```
... either as the default slot content:
```xml
<t t-component="commandItem.Component" t-props="commandItem.props">
    <t t-esc="commandItem.name"/>
</t>
```

I let you decide on that, but there is my opinion :-)